ui/src/sources/components/SourceStep.tsx

```diff
@@ -159,6 +160,14 @@ class SourceStep extends PureComponent<Props, State> {
     )
   }
 
+  private get passwordPlaceholder() {
+    const {source} = this.props
+    if (source.authentication === 'basic') {
+      return 'Value saved in server'
+    }
+    return null
```

Comment:
How do you feel about removing the `return null` here? it isn't necessary since the function will return undefined if it does not specifically return.